File: 프로그래머스/Level1/3진법 뒤집기.py
# https://programmers.co.kr/learn/courses/30/lessons/68935

def solution(n):
    def toThree(n):
        temp = ''
        while n > 0:
            res = n % 3
            n = n // 3
            temp += str(res)
        return temp[::-1]
    temp = toThree(n)
    answer = 0
    b = 1
    for a in temp:
        answer += int(a) * b
        b *= 3
    return answer

File: 프로그래머스/Level1/3진법 뒤집기.js
// https://programmers.co.kr/learn/courses/30/lessons/68935

const three = (n) => {
	let result = '';
	while (n != 0) {
		const res = n % 3;
		n = parseInt(n / 3);
		result = res + result;
	}
	return result;
};

const solution = (n) => {
	const a = three(n);
	return a
		.split('')
		.map((num, idx) => 3 ** idx * num)
		.reduce((prev, cur) => prev + cur, 0);
};
